litellm/proxy/auth/auth_checks.py

```diff
@@ -692,8 +693,19 @@ async def get_user_object(
 
         if response is None:
             if user_id_upsert:
+                # Create a new user with the specified role
```

Comment:
you just need to use litellm.default_internal_user_params here https://docs.litellm.ai/docs/proxy/self_serve#all-settings-for-self-serve--sso-flow